refactored lifecycle hooks

iamritikiit opened the pull request:
<!-- Thanks for contributing to Cuttle! 🎉 -->

## Issue number

Relevant [issue number](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- Resolves #

## Please check the following

- [ ] Do the tests still pass? (see [Run the Tests](https://github.com/cuttle-cards/cuttle/blob/main/docs/setup-development.md#run-the-tests))
- [ ] Is the code formatted properly? (see [Linting (Formatting)](https://github.com/cuttle-cards/cuttle/blob/main/docs/setup-development.md#linting-formatting))
- For New Features:
  - [ ] Have tests been added to cover any new features or fixes?
  - [ ] Has the documentation been updated accordingly?

## Please describe additional details for testing this change

Haviles04 commented on src/routes/login/LoginView.vue:
```diff
@@ -300,7 +300,7 @@ export default {
     }
   },
-  mounted() {
+  created() {
```
I think this one should stay on mounted since it's scrolling to something in the DOM

iamritikiit replied:
Thanks for catching that! I’ve moved the scrolling logic back to mounted so it runs after the DOM is available.

Haviles04 commented on src/routes/home/components/announcementDialog/AnnouncementDialog.vue:
```diff
@@ -80,11 +80,12 @@ const close = () => {
-    show.value = true;
-  }
-});
+// Initialisation logic moved from onMounted
```
No need for comments, if we're going to remove it we can just remove it

iamritikiit replied:
Sure, I have now removed the unnecessary comments.

Haviles04 reviewed (changes requested):
Thanks for your contribution! I left a few small points of feedback :)

itsalaidbacklife commented on src/routes/login/LoginView.vue:
```diff
@@ -300,7 +300,7 @@ export default {
     }
   },
-  mounted() {
+  created() {
```
This needs to stay mounted because the function is interacting with the DOM (scrolling)
```suggestion
  mounted() {
```

iamritikiit replied:
Thanks for pointing that out, I have now moved the logic back to mounted so that it runs after DOM is available.

itsalaidbacklife commented on src/routes/lobby/LobbyView.vue:
```diff
@@ -179,9 +179,9 @@ watch(opponentUsername, (newVal) => {
-onMounted(() => {
-  playAudio(joinAudio);
-});
+// onMounted(() => {
+playAudio(joinAudio);
+// });
```
This is a sneaky one but it needs to stay in onMounted because playAudio() is interacting with the DOM 
```suggestion
 onMounted(() => {
playAudio(joinAudio);
 });
```

iamritikiit replied:
Yes, we should keep playAudio() in onMounted as this will be interacting with the browser's DOM and media APIs.
Now, I have moved back the logic inside of the onMounted() again.

itsalaidbacklife reviewed (changes requested):
Thank you for your contribution! This is a good start. I've requested several changes. I haven't yet scanned the rest of the code base to verify if this is comprehensive; have you searched for all the mounted and onMounted calls throughout the project?

iamritikiit commented on src/routes/home/components/announcementDialog/AnnouncementDialog.vue:
```diff
@@ -80,11 +80,12 @@ const close = () => {
-  }
-});
+// Initialisation logic moved from onMounted
+// onMounted(() => { 
```
Yes, I have searched for all the mounted and onMounted calls throughout the project and made the required changes, also have removed the unnecessary comments.

iamritikiit commented:
I’ve addressed all the feedback and resolved the previous comments. Please let me know if any further adjustments are needed. Looking forward to your review!

Haviles04 reviewed (approved):
This lgtm, I searched for all mounted and onMounted calls and I think all the ones we need to change have been addressed
